zookeeper-server/src/test/java/org/apache/zookeeper/server/admin/CommandsTest.java

```diff
@@ -193,8 +193,15 @@ public void testMonitor() throws IOException, InterruptedException {
                 new Field("global_sessions", Long.class),
                 new Field("local_sessions", Long.class),
                 new Field("connection_drop_probability", Double.class)
-        ));
-        for (String metric : ServerMetrics.getAllValues().keySet()) {
+        ));        
+        Map<String, Object> metrics = new HashMap<>();
+        ServerMetrics
+                .getMetrics()
+                .getMetricsProvider()
+                .dump( (metric, value)-> {
+                    metrics.put(metric, value);
+                });
+        for (String metric : metrics.keySet()) {
```

Comment:
nit: could remove this for loop and put into the dump callback.

Reply:
Do you think it's worth to add a method in the metric provider to return a map, since we saw this pattern happened multiple times?

Reply:
@lvfangmin I am not sure it is worth.
This code is only for tests.

I don't want to force a real 'provider' to implement such API.

The dump API does not force the creation of a Map, which will implicitly require cpu cycles and allocations.

I thought about adding a 'default' method but I think it could be abused, that is, used in production code (like the monitor command or an http endpoint)  and not in tests.

I can provide an utility for tests inside the tests package.
How does this sounds to you?

Reply:
@lvfangmin

Now we are using your default implementation even in tests.
I think a better approach for the mid term  future will be to create a TestMetricsProvider which gives access to its internals and so it is more useful for mocks and for advanced unit testing

Reply:
Adding a utility method in test sounds good to me, having TestMetricsProvider sounds better for future testing purpose.

Reply:
@lvfangmin I have added a MetricUtils in order to collect all of the current Metrics, thanks for your suggestion.
I am not introducing now TestMetricsProvider because the patch will be bigger and we could change the meaning of some tests, so I prefer to keep the patch as small as possible.